app/actors/hyrax/actors/apply_permission_template_actor.rb

```diff
@@ -13,8 +13,23 @@ def create(env)
       private
 
         def add_edit_users(env)
-          return if env.attributes[:source_id].blank?
-          template = Hyrax::PermissionTemplate.find_by!(source_id: env.attributes[:source_id])
+          add_admin_set_participants(env)
+          add_collection_participants(env)
+        end
+
+        def add_admin_set_participants(env)
+          return if env.attributes[:admin_set_id].blank?
+          template = Hyrax::PermissionTemplate.find_by!(source_id: env.attributes[:admin_set_id])
+          env.curation_concern.edit_users += template.agent_ids_for(agent_type: 'user', access: 'manage')
+          env.curation_concern.edit_groups += template.agent_ids_for(agent_type: 'group', access: 'manage')
+          env.curation_concern.read_users += template.agent_ids_for(agent_type: 'user', access: 'view')
+          env.curation_concern.read_groups += template.agent_ids_for(agent_type: 'group', access: 'view')
+        end
+
+        def add_collection_participants(env)
+          return if env.attributes[:collection_id].blank?
+          collection_id = env.attributes.delete(:collection_id)
```

Comment:
Why is the `collection_id` parameter being deleted and the `admin_set_id` is not being deleted? Please add a comment on why delete is used here and not for the `admin_set_id`.

Reply:
When the actor saving generic_work saves, it can have the admin_set_id in the attributes because works responds to a method of that name.  Works do not respond to a method named :collection_id.  So it must be removed.

I am assuming that no only does a work respond to admin_set_id, but it is expecting it to be part of the attributes.

I'll add a comment.